Sources/MongoSwift/MongoCollection.swift

```diff
@@ -133,4 +133,92 @@ public class MongoCollection<T: Codable> {
+     *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination or the pipeline passed
+     *     is invalid.
+     *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+     *     pipeline.
+     */
+    public func watch(_ pipeline: [Document],
+                      options: ChangeStreamOptions? =  nil,
+                      session: ClientSession? = nil) throws ->
+                      ChangeStream<ChangeStreamDocument<CollectionType>> {
+        return try self.watch(pipeline, options: options, session: session, withFullDocumentType: CollectionType.self)
+    }
+
+    /**
+     * Starts a `ChangeStream` on a collection. Associates the specified `Codable` type `T` with the `fullDocument`
+     * field in the `ChangeStreamDocument`.
+     * - Parameters:
+     *   - Pipeline: The pipeline of stages to append to an initial `ChangeStream` stage.
+     *   - Options: An optional `ChangeStreamOptions` to use on the initial `ChangeStream` stage.
+     *   - Session: An optional `ChangeStream` to use with this change stream.
+     * - Returns: A change stream on a specific collection.
+     * - Throws:
+     *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+     *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination or the pipeline passed
+     *     is invalid.
+     *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+     *     pipeline.
+     */
+    public func watch<T: Codable>(_ pipeline: [Document],
+                                  options: ChangeStreamOptions? = nil,
+                                  session: ClientSession? = nil,
+                                  withFullDocumentType type: T.Type) throws ->
+                                  ChangeStream<ChangeStreamDocument<T>> {
+        let pipeline: Document = ["pipeline": pipeline]
+        let connection = try self._client.connectionPool.checkOut()
+        return try self.withMongocCollection(from: connection) { collPtr in
+            let opts = try encodeOptions(options: options, session: session)
+            let changeStreamPtr: OpaquePointer = mongoc_collection_watch(collPtr, pipeline._bson, opts?._bson)
+            return try ChangeStream<ChangeStreamDocument<T>>(stealing: changeStreamPtr,
+                                                             client: self._client,
+                                                             connection: connection,
+                                                             session: session,
+                                                             decoder: self.decoder)
+        }
+    }
+
+    /**
+     * Starts a `ChangeStream` on a collection. Associates the specified `Codable` type `T` with the returned
+     * `ChangeStream`.
+     * - Parameters:
```

Comment:
Let's add here (and for all the `withXType` parameters) descriptions of what they're doing. In this case, we could say something like "The type that the change events emitted from the change stream will be decoded to". 

I also think it's worth pointing out in the "Throws" section that for methods in which a user can specify a `Codable` type, a `DecodingError` will be thrown if the decoding doesn't work. e.g. if I pass in `struct { let nonsense: Int }` as the return type, I'll get an error.

Reply:
Yeah, unless I overlooked, `mongoc_watch` does not call the server only creates the cursor and the command to be sent later.
Removed `ServerError` from the docstrings.

Reply:
I think it does talk to the server. `mongoc_client_watch` calls `_mongoc_change_stream_new_from_client`, which calls `_change_stream_init`, which calls `_make_cursor`, which calls `mongoc_client_read_command_with_opts` and sends a read command. 
see [here](https://github.com/mongodb/mongo-c-driver/blob/2d7d5f20261eb797afed22a17c397c8c91014605/src/libmongoc/src/mongoc/mongoc-change-stream.c#L232)

docstring for `make_cursor` also says "Construct and send the aggregate command and create the resulting cursor."

(you could also verify this with command monitoring)

note this is different from the behavior of methods that return `MongoCursor`s. all of those are lazy and don't send the initial cursor command until you try to read from the cursor.

Reply:
Oops, I see that now. Also, when I deleted my comment about not seeing the call to the server (realized it was in the wrong place), I had not seen your response yet. 😅